From 69050f8d6d075dc01af7a5f2f550a8067510366f Mon Sep 17 00:00:00 2001
From: Kees Cook <kees@kernel.org>
Date: Fri, 20 Feb 2026 23:49:23 -0800
Subject: [PATCH] treewide: Replace kmalloc with kmalloc_obj for non-scalar
 types

This is the result of running the Coccinelle script from
scripts/coccinelle/api/kmalloc_objs.cocci. The script is designed to
avoid scalar types (which need careful case-by-case checking), and
instead replace kmalloc-family calls that allocate struct or union
object instances:

Single allocations:	kmalloc(sizeof(TYPE), ...)
are replaced with:	kmalloc_obj(TYPE, ...)

Array allocations:	kmalloc_array(COUNT, sizeof(TYPE), ...)
are replaced with:	kmalloc_objs(TYPE, COUNT, ...)

Flex array allocations:	kmalloc(struct_size(PTR, FAM, COUNT), ...)
are replaced with:	kmalloc_flex(*PTR, FAM, COUNT, ...)

(where TYPE may also be *VAR)

The resulting allocations no longer return "void *", instead returning
"TYPE *".

Signed-off-by: Kees Cook <kees@kernel.org>
---
 drivers/media/platform/rockchip/rkvdec/rkvdec-h264.c         | 2 +-
 drivers/media/platform/rockchip/rkvdec/rkvdec-hevc.c         | 2 +-
 drivers/media/platform/rockchip/rkvdec/rkvdec-vdpu381-h264.c | 2 +-
 drivers/media/platform/rockchip/rkvdec/rkvdec-vdpu381-hevc.c | 2 +-
 drivers/media/platform/rockchip/rkvdec/rkvdec-vdpu383-h264.c | 2 +-
 drivers/media/platform/rockchip/rkvdec/rkvdec-vdpu383-hevc.c | 2 +-
 drivers/media/platform/rockchip/rkvdec/rkvdec-vp9.c          | 2 +-
 drivers/media/platform/rockchip/rkvdec/rkvdec.c              | 2 +-
 drivers/media/platform/verisilicon/hantro_drv.c              | 2 +-
 9 files changed, 9 insertions(+), 9 deletions(-)

--- a/drivers/media/platform/rockchip/rkvdec/rkvdec-h264.c
+++ b/drivers/media/platform/rockchip/rkvdec/rkvdec-h264.c
@@ -375,7 +375,7 @@ static int rkvdec_h264_start(struct rkvd
 	if (ret)
 		return ret;
 
-	h264_ctx = kzalloc(sizeof(*h264_ctx), GFP_KERNEL);
+	h264_ctx = kzalloc_obj(*h264_ctx, GFP_KERNEL);
 	if (!h264_ctx)
 		return -ENOMEM;
 
--- a/drivers/media/platform/rockchip/rkvdec/rkvdec-hevc.c
+++ b/drivers/media/platform/rockchip/rkvdec/rkvdec-hevc.c
@@ -528,7 +528,7 @@ static int rkvdec_hevc_start(struct rkvd
 	struct rkvdec_hevc_priv_tbl *priv_tbl;
 	struct rkvdec_hevc_ctx *hevc_ctx;
 
-	hevc_ctx = kzalloc(sizeof(*hevc_ctx), GFP_KERNEL);
+	hevc_ctx = kzalloc_obj(*hevc_ctx, GFP_KERNEL);
 	if (!hevc_ctx)
 		return -ENOMEM;
 
--- a/drivers/media/platform/rockchip/rkvdec/rkvdec-vdpu381-h264.c
+++ b/drivers/media/platform/rockchip/rkvdec/rkvdec-vdpu381-h264.c
@@ -383,7 +383,7 @@ static int rkvdec_h264_start(struct rkvd
 	if (ret)
 		return ret;
 
-	h264_ctx = kzalloc(sizeof(*h264_ctx), GFP_KERNEL);
+	h264_ctx = kzalloc_obj(*h264_ctx, GFP_KERNEL);
 	if (!h264_ctx)
 		return -ENOMEM;
 
--- a/drivers/media/platform/rockchip/rkvdec/rkvdec-vdpu381-hevc.c
+++ b/drivers/media/platform/rockchip/rkvdec/rkvdec-vdpu381-hevc.c
@@ -551,7 +551,7 @@ static int rkvdec_hevc_start(struct rkvd
 	if (ret)
 		return ret;
 
-	hevc_ctx = kzalloc(sizeof(*hevc_ctx), GFP_KERNEL);
+	hevc_ctx = kzalloc_obj(*hevc_ctx, GFP_KERNEL);
 	if (!hevc_ctx)
 		return -ENOMEM;
 
--- a/drivers/media/platform/rockchip/rkvdec/rkvdec-vdpu383-h264.c
+++ b/drivers/media/platform/rockchip/rkvdec/rkvdec-vdpu383-h264.c
@@ -447,7 +447,7 @@ static int rkvdec_h264_start(struct rkvd
 	if (ret)
 		return ret;
 
-	h264_ctx = kzalloc(sizeof(*h264_ctx), GFP_KERNEL);
+	h264_ctx = kzalloc_obj(*h264_ctx, GFP_KERNEL);
 	if (!h264_ctx)
 		return -ENOMEM;
 
--- a/drivers/media/platform/rockchip/rkvdec/rkvdec-vdpu383-hevc.c
+++ b/drivers/media/platform/rockchip/rkvdec/rkvdec-vdpu383-hevc.c
@@ -559,7 +559,7 @@ static int rkvdec_hevc_start(struct rkvd
 	if (ret)
 		return ret;
 
-	hevc_ctx = kzalloc(sizeof(*hevc_ctx), GFP_KERNEL);
+	hevc_ctx = kzalloc_obj(*hevc_ctx, GFP_KERNEL);
 	if (!hevc_ctx)
 		return -ENOMEM;
 
--- a/drivers/media/platform/rockchip/rkvdec/rkvdec-vp9.c
+++ b/drivers/media/platform/rockchip/rkvdec/rkvdec-vp9.c
@@ -972,7 +972,7 @@ static int rkvdec_vp9_start(struct rkvde
 	unsigned char *count_tbl;
 	int ret;
 
-	vp9_ctx = kzalloc(sizeof(*vp9_ctx), GFP_KERNEL);
+	vp9_ctx = kzalloc_obj(*vp9_ctx, GFP_KERNEL);
 	if (!vp9_ctx)
 		return -ENOMEM;
 
--- a/drivers/media/platform/rockchip/rkvdec/rkvdec.c
+++ b/drivers/media/platform/rockchip/rkvdec/rkvdec.c
@@ -1282,7 +1282,7 @@ static int rkvdec_open(struct file *filp
 	struct rkvdec_ctx *ctx;
 	int ret;
 
-	ctx = kzalloc(sizeof(*ctx), GFP_KERNEL);
+	ctx = kzalloc_obj(*ctx, GFP_KERNEL);
 	if (!ctx)
 		return -ENOMEM;
 
--- a/drivers/media/platform/verisilicon/hantro_drv.c
+++ b/drivers/media/platform/verisilicon/hantro_drv.c
@@ -640,7 +640,7 @@ static int hantro_open(struct file *filp
 	 * helper functions used here.
 	 */
 
-	ctx = kzalloc(sizeof(*ctx), GFP_KERNEL);
+	ctx = kzalloc_obj(*ctx, GFP_KERNEL);
 	if (!ctx)
 		return -ENOMEM;
 
